man/man7/git-secret.7.ronn

```diff
@@ -49,13 +58,22 @@ After doing so rerun the tests to be sure that it won't break anything. Tested t
 
 * `$SECRETS_EXTENSION` - sets the secret files extension, defaults to `.secret`. It can be changed to any valid file extension.
 
-## Internals -- the `.gitsecret` folder
+## The `.gitsecret` folder
 
-This folder contains all the information about the data encrypted in this repo, 
+This folder contains information about the files encrypted by git-secret, 
 and about which public/private key sets can access the encrypted data. 
+
 Use the various 'git secret' commands to manipulate the files in `.gitsecret`, 
 you should not change the data in these files directly.
 
+Exactly which files exist in the .gitsecret folder and what their contents are
+vary slightly across different versions of gpg. Thus it is best to use
+git-secret with versions of gpg that interoperate well.
```

Comment:
Hm. What versions of gpg should we specify? 

My first reaction would be to say "use a recent version of gpg 2.2", but some operating systems ship with older versions of gpg (centos 7 has 2.0.22) or are hard to upgrade from other versions.

Reply:
And `gpg@1` is also still popular.

Reply:
I guess, we should specify versions that we *recommend*. But versions that we really support might differ. `gpg@2.2` is fine.

Reply:
Generally, everything is fine you use git-secret on just one version of gpg. 

The issue is when people use git-secret on different systems (say, mac and linux), and the .gitsecret/ files are changed on both systems. Since the contents of .gitsecret/keys are dependent on the version of gpg used, if you make changes to a git-secret repo from different systems, the way they manipulate files in .gitsecret/keys might not interoperate well. 

Here's the versions of gpg that are most easily available on some systems:
* osx - gnupg 2.2.7 via brew
* centos - gnupg 2.0.22 via rpm
* debian latest - gnupg (2.1.18-8~deb9u2) via deb

Perhaps we should provide an interoperability matrix, showing which common versions of gpg are known to interoperate well or poorly.

Reply:
Your point about 'supported' version(s) is a good idea too @sobolevn

Reply:
This now reads:
````
Exactly which files exist in the .gitsecret folder and what their contents are
vary slightly across different versions of gpg. Thus it is best to use
git-secret with the same version of gpg being used by all users.
This can be forced using SECRETS_GPG_COMMAND environment variable.
````